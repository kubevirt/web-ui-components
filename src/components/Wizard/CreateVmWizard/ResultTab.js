import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'patternfly-react';
import { Loading } from '../../Loading';

const ResultTab = ({ result, success }) => {
  let value;
  if (success == null) {
    value = <Loading text="Creation of VM in progress" />;
  } else if (success) {
    value = (
      <div className="wizard-pf-complete blank-slate-pf" key="success">
        <div className="wizard-pf-success-icon">
          <span className="glyphicon glyphicon-ok-circle" />
        </div>
        <h3 className="blank-slate-pf-main-action">Creation of VM was successful</h3>
        <pre className="blank-slate-pf-secondary-action description">{result}</pre>
      </div>
    );
  } else {
    value = <Alert key="fail">{result}</Alert>;
  }

  return value;
};

ResultTab.defaultProps = {
  result: null,
  success: null,
};

ResultTab.propTypes = {
  result: PropTypes.string,
  success: PropTypes.bool,
};

export default ResultTab;
