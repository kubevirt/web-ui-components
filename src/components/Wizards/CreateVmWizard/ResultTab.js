import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'patternfly-react';

const ResultTab = ({ result, success }) => {
  let value;
  if (success == null) {
    value = (
      <div className="wizard-pf-process blank-slate-pf" key="progress">
        <div className="spinner spinner-lg blank-slate-pf-icon" />
        <h3 className="blank-slate-pf-main-action">Creation of VM in progress</h3>
      </div>
    );
  } else if (success) {
    value = (
      <div className="wizard-pf-complete blank-slate-pf" key="success">
        <div className="wizard-pf-success-icon">
          <span className="glyphicon glyphicon-ok-circle" />
        </div>
        <h3 className="blank-slate-pf-main-action">Creation of VM was succesfull</h3>
        <pre className="blank-slate-pf-secondary-action description">{result}</pre>
      </div>
    );
  } else {
    value = <Alert key="fail">{result}</Alert>;
  }

  return value;
};

ResultTab.defaultProps = {
  result: null,
  success: null
};

ResultTab.propTypes = {
  result: PropTypes.string,
  success: PropTypes.bool
};

export default ResultTab;
