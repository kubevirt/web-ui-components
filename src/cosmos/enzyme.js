import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
