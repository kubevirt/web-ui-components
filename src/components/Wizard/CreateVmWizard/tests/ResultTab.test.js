import React from 'react';
import { shallow } from 'enzyme';

import { ResultTab } from '../ResultTab';

describe('<ResultTab />', () => {
  it('renders progress correctly', () => {
    const component = shallow(<ResultTab success={null} result={null} />);
    expect(component).toMatchSnapshot();
  });

  it('renders success correctly', () => {
    const component = shallow(<ResultTab success result="Created VM" />);
    expect(component).toMatchSnapshot();
  });

  it('renders fail correctly', () => {
    const component = shallow(<ResultTab success={false} result="Failed to create VM" />);
    expect(component).toMatchSnapshot();
  });
});
