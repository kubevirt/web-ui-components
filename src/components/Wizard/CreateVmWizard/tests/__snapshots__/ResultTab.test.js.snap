// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultTab /> renders fail correctly 1`] = `
<Alert
  className=""
  key="fail"
  onDismiss={null}
  type="error"
>
  Failed to create VM
</Alert>
`;

exports[`<ResultTab /> renders progress correctly 1`] = `
<Loading
  text="Creation of VM in progress"
/>
`;

exports[`<ResultTab /> renders success correctly 1`] = `
<div
  className="wizard-pf-complete blank-slate-pf"
  key="success"
>
  <div
    className="wizard-pf-success-icon"
  >
    <span
      className="glyphicon glyphicon-ok-circle"
    />
  </div>
  <h3
    className="blank-slate-pf-main-action"
  >
    Creation of VM was successful
  </h3>
  <pre
    className="blank-slate-pf-secondary-action description"
  >
    Created VM
  </pre>
</div>
`;
